src/Sentry/Protocol/SentryAttribute.cs

```diff
@@ -0,0 +1,107 @@
+namespace Sentry.Protocol;
+
+[DebuggerDisplay(@"\{ Value = {Value}, Type = {Type} \}")]
+internal readonly struct SentryAttribute
+{
+    public SentryAttribute(object value, string type)
+    {
+        Value = value;
+        Type = type;
+    }
+
+    public object Value { get; }
+    public string Type { get; }
+}
+
+internal static class SentryAttributeSerializer
+{
+    internal static void WriteAttribute(Utf8JsonWriter writer, string propertyName, SentryAttribute attribute)
+    {
+        Debug.Assert(attribute.Value is not null && attribute.Type is not null, $"The ValueType {nameof(attribute)} may have been assigned 'default', for which static flow analysis does not report nullable warnings.");
+        writer.WritePropertyName(propertyName);
+        WriteAttributeValue(writer, attribute.Value, attribute.Type);
+    }
+
+    internal static void WriteAttribute(Utf8JsonWriter writer, string propertyName, object value, string type)
+    {
+        writer.WritePropertyName(propertyName);
+        WriteAttributeValue(writer, value, type);
+    }
+
+    internal static void WriteAttribute(Utf8JsonWriter writer, string propertyName, object value)
+    {
+        writer.WritePropertyName(propertyName);
+        WriteAttributeValue(writer, value);
+    }
+
+    private static void WriteAttributeValue(Utf8JsonWriter writer, object value, string type)
+    {
+        writer.WriteStartObject();
+
+        if (type == "string")
+        {
+            writer.WriteString("value", (string)value);
+            writer.WriteString("type", type);
+        }
+        else if (type == "boolean")
+        {
+            writer.WriteBoolean("value", (bool)value);
+            writer.WriteString("type", type);
+        }
+        else if (type == "integer")
+        {
+            writer.WriteNumber("value", (long)value);
+            writer.WriteString("type", type);
+        }
+        else if (type == "double")
+        {
+            writer.WriteNumber("value", (double)value);
+            writer.WriteString("type", type);
+        }
+        else
```

Comment:
Is this expected? If not, should we debug log?
Will be worse to get some reports that "I'm getting Namespace.MyType in the logs"?

Reply:
while adding more tests for this case, I discovered the possibility of Null-Reference-Exception ... fixed and added Log messages when passing unsupported types